File: app/src/main/java/com/example/groceryapp/groceryDb.java
package com.example.groceryapp;


import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.groceryapp.model.Record1;

@Database(entities = Record1.class, version = 12)
public abstract class groceryDb extends RoomDatabase {
    private static groceryDb groceryDatabase = null;
    private static final String databaseName = "groceryRecords";

    public abstract groceryDao groceryDao();

    public static synchronized groceryDb getGroceryDatabase(Context context)
    {
        if(groceryDatabase == null)
            groceryDatabase = Room.databaseBuilder(context.getApplicationContext(),groceryDb.class,databaseName).
                    allowMainThreadQueries().build();

        return groceryDatabase;
    }


}

File: app/src/main/java/com/example/groceryapp/Room/Grocerydb.kt
//package com.example.groceryapp.Room
//
//import android.content.Context
//import androidx.room.Database
//import androidx.room.Room
//import androidx.room.RoomDatabase
//
//
//
//@Database(entities = [ GroceryModel::class], version = 1)
//
//public abstract  class Grocerydb: RoomDatabase() {
//
//    abstract fun GroceryDao(): Grocerydao?
// public var Grocerydatabase:Grocerydb?=null
//val context: Context?=null
//
//    @Synchronized
//  public  fun GroceryDatabase(ctx:Context): Grocerydb?
//    {
//        if(Grocerydatabase!= null) {
//            Grocerydatabase= Room.databaseBuilder(ctx.applicationContext,Grocerydb::class.java,"grocery_db")
//                .fallbackToDestructiveMigration()
//                .build()
//        }
//
//
//
//        return Grocerydatabase
//
//    }
//
//}